app/src/main/java/clarkson/ee408/tictactoev4/LoginActivity.java

```diff
@@ -40,31 +49,56 @@ public void handleLogin() {
         String username = usernameField.getText().toString();
         String password = passwordField.getText().toString();
 
-        // TODO: verify that all fields are not empty before proceeding. Toast with the error message
+        // Verify that all fields are not empty before proceeding. Toast with the error message
+        if (username.isEmpty() || password.isEmpty()) {
+            Toast.makeText(this, "Please enter both username and password", Toast.LENGTH_SHORT).show();
+            return;
+        }
 
-        // TODO: Create User object with username and password and call submitLogin()
+        // Create User object with username and password and call submitLogin()
+        User user = new User(username, password);
```

Comment:
There isn't a constructor for User with only two parameters. Use the default constructor and setters

```suggestion
        User user = new User();
        user.setUsername(username);
        user.setPassword(password);
```